core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -2131,12 +2202,12 @@ class LogTest {
       assertEquals("Should have same number of time index entries as before.", numTimeIndexEntries, log.activeSegment.timeIndex.entries)
     }
 
-    log = createLog(logDir, logConfig, recoveryPoint = lastOffset)
+    log = createLog(logDir, logConfig, recoveryPoint = lastOffset, lastShutdownClean = false)
```

Comment:
It seems that this expects a clean shutdown.

Reply:
Jun, this test was not creating a clean shut down file before opening the log again. So, it would have gone through the recovery path. Hence, I have set `lastShutdownClean` parameter to `false`. Similarly, for line 2210.